scapy/layers/l2.py

```diff
@@ -807,33 +809,45 @@ def arp_mitm(
         $ sudo scapy
         >>> arp_mitm("192.168.122.156", "192.168.122.17")
 
+    Alternative usages:
+        >>> arp_mitm("10.0.0.1", "10.1.1.0/21", iface="eth1")
+        >>> arp_mitm("10.0.0.1", "10.1.1.2",
+        ...          target_mac="aa:aa:aa:aa:aa:aa",
+        ...          mac2="00:1e:eb:bf:c1:ab")
+
     Remember to change the sysctl settings back..
     """
     if not iface:
         iface = conf.route.route(ip1)[0]
     if not target_mac:
         target_mac = get_if_hwaddr(iface)
     if mac1 is None:
-        mac1 = getmacbyip(ip1)
-        if not mac1:
-            print("Can't resolve mac for %s" % ip1)
-            return
+        if "/" in ip1:  # subnet
+            mac1 = "ff:ff:ff:ff:ff:ff"
```

Comment:
I'm not sure you should do that, since that means the "real" hosts behind `ip1` will receive the ARP packet and hence detect an "IP address conflict". This will cause troubles.

Another option could be to do something like
```Python
ip1 = [ip for ip in Net(ip1)]
mac1 = [getmacbyip(ip) for ip in ip1]
```

and then send packets only to IP addresses for which we have a MAC address?

Same for `ip2` of course

Reply:
That's a valid point

Reply:
I used `arping` instead of `getmacbyip` for performance. What do you think of the change?

Reply:
I like that but be careful! With your current code, `macX` are lists, so when you `sendp(Ether(dst=macX)/ARP(pdst=ipX)` you will send a packet for each IP in `ipX` to each MAC in `macX`, which is not what you want!
You should keep the association `(mac, ip)` from the result of `arping()` and use it to send only the correct packets.